packages/frontend/core/src/components/page-list/tags/create-tag.tsx

```diff
@@ -43,14 +38,16 @@ export const CreateOrEditTag = ({
     setTagName(value);
   }, []);
 
-  const [tagIcon, setTagIcon] = useState(tagMeta?.color || randomTagColor());
+  const [tagIcon, setTagIcon] = useState(
+    tagMeta?.color || tagService.randomTagColor()
+  );
 
   const handleChangeIcon = useCallback((value: string) => {
     setTagIcon(value);
   }, []);
 
   const tags = useMemo(() => {
-    return tagColors.map(([_, color]) => {
+    return tagService.tagColors.map(([_, color]) => {
```

Comment:
`tagColors` better to be a map?

Reply:
Sometimes the code will rely on the order of this tagColors array

https://github.com/toeverything/AFFiNE/blob/715187f90f3726540d3381015b056a2220a3c42e/packages/frontend/core/src/components/affine/page-properties/tags-inline-editor.tsx#L271

Reply:
crazy but map/set in js are ordered